### ISSUE TITLE
Distortion coefficients

### ISSUE THREAD

**Issue by MaxTeselkin** [facebookresearch/vggt#46]
Hi! I am trying to use VGGT for camera calibration in order to project points of 3D point cloud to 2D photo context image. I have extracted extrinsic and intrinsic matrices from VGGT, but it is not enough for projecting points on 2D image - it is also necessary to get distortion coefficients. I tried to set distortion coefficients to 0, but it didn't work. Here is my code:

```python
# get lidar points
pcd = o3d.io.read_point_cloud(local_pcd_path, format="pcd")
pcd_points = np.asarray(pcd.points)

device = "cuda" if torch.cuda.is_available() else "cpu"
# bfloat16 is supported on Ampere GPUs (Compute Capability 8.0+) 
dtype = torch.bfloat16 if torch.cuda.get_device_capability()[0] >= 8 else torch.float16

# Initialize the model and load the pretrained weights.
# This will automatically download the model weights the first time it's run, which may take a while.
model = VGGT.from_pretrained("facebook/VGGT-1B").to(device)

# Load and preprocess example images (replace with your own image paths)
image_names = ["../experiment_data/photo_context.jpg"]  
images = load_and_preprocess_images(image_names).to(device)

with torch.no_grad():
    with torch.cuda.amp.autocast(dtype=dtype):
        images = images[None]  # add batch dimension
        aggregated_tokens_list, ps_idx = model.aggregator(images)
                
    # Predict Cameras
    pose_enc = model.camera_head(aggregated_tokens_list)[-1]
    # Extrinsic and intrinsic matrices, following OpenCV convention (camera from world)
    extrinsic_matrix, intrinsic_matrix = pose_encoding_to_extri_intri(pose_enc, images.shape[-2:])
    extrinsic_matrix = extrinsic_matrix.detach().cpu().numpy().squeeze()
    intrinsic_matrix = intrinsic_matrix.detach().cpu().numpy().squeeze()

# extract rotation vector from extrinsic matrix
rotation_matrix = extrinsic_matrix[:, :3]

# extract translation vector from extrinsic matrix
translation_vector = extrinsic_matrix[:, 3]

# set distortion coefficients to 0
dist_coeffs = np.zeros(5)

# get point projections
point_projections, _ = cv2.projectPoints(
    pcd_points,
    rotation_matrix,
    translation_vector,
    intrinsic_matrix,
    dist_coeffs,
)
```

Is there a way to get distortion coefficients from VGGT?

**Comment by jytime** [comment]
Hi,

Our model assumes zero distortion, meaning it expects input images to be either undistorted or have minimal distortion. Thus, using **zero distortion coefficients** should generally work reasonably well.

However, if you’re still facing issues, here are two points to double-check:

### 1. Coordinate System
Ensure your LiDAR points are correctly transformed into the **OpenCV coordinate system** before projection.

### 2. Scale and Shift
Our model predicts camera poses within a **normalized coordinate system**. Your LiDAR points may have a different scale and shift. To align them:

- Compute a global scale and shift transformation by aligning a subset of LiDAR points to our predicted points.
- Apply this transformation to your entire point cloud.

You don’t need perfect alignment—just select some corresponding points visible in both LiDAR and image data to calculate this transform.

Let me know if you have further questions!
